core/citrus-api/src/main/java/com/consol/citrus/spi/ResourcePathTypeResolver.java

```diff
@@ -130,6 +143,37 @@ public <T> Map<String, T> resolveAll(String resourcePath, String property, Strin
         return resources;
     }
 
+    private List<Resource> resolveAllFromJar(String path) {
+        List<Resource> result = new ArrayList<>();
```

Comment:
if you would return the stream starting at `getZipEntries().stream()` and then use lambdas to do the mapping and filtering accordingly you could event skip this resource allocation. did you think about that?

Reply:
I am not sure what exactly you mean. Do you mean returning a `Stream<Resource>` instead of a `List<Resource>`?

Reply:
@bbortt okay I think I get what you meant. Updated the PR.

Reply:
that is exactly what I meant! 😉 could you test this with a local installation?

Reply:
Already did. Works fine.